miio/airhumidifier_jsq.py

```diff
@@ -0,0 +1,330 @@
+        return self.data["child_lock"] == 1
+
+    @property
+    def no_water(self) -> bool:
+        """True if the water tank is empty."""
+        return self.data["no_water"] == 1
+
+    @property
+    def lid_opened(self) -> bool:
+        """True if the water tank is detached."""
+        return self.data["lid_opened"] == 1
+
+    @property
+    def firmware_version(self) -> str:
+        """Returns the fw_ver of miIO.info. For example 1.3.9."""
+        return self.device_info.firmware_version
+
+    @property
+    def hardware_version(self) -> Optional[str]:
+        """The hardware version."""
+        return self.device_info.hardware_version
+
+    def __repr__(self) -> str:
+        s = (
+            "<AirHumidiferStatus power=%s, "
+            "mode=%s, "
+            "temperature=%s, "
+            "humidity=%s%%, "
+            "led_brightness=%s, "
+            "buzzer=%s, "
+            "child_lock=%s, "
+            "no_water=%s, "
+            "lid_opened=%s, "
+            "hardware_version=%s, "
+            "firmware_version=%s>"
+            % (
+                self.power,
+                self.mode,
+                self.temperature,
+                self.humidity,
+                self.led_brightness,
+                self.buzzer,
+                self.child_lock,
+                self.no_water,
+                self.lid_opened,
+                self.firmware_version,
+                self.hardware_version,
+            )
+        )
+        return s
+
+    def __json__(self):
+        return self.data
+
+
+class AirHumidifierJsq(Device):
+    """
+    Implementation of Xiaomi The fog free of humidifier:
+      shuii.humidifier.jsq001
+     """
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_HUMIDIFIER_JSQ001,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in AVAILABLE_PROPERTIES:
+            self.model = model
+        else:
+            self.model = MODEL_HUMIDIFIER_JSQ001
+
+        self.device_info = None
+
+    @command(
+        default_output=format_output(
+            "",
+            "Power: {result.power}\n"
+            "Mode: {result.mode}\n"
+            "Temperature: {result.temperature} °C\n"
+            "Humidity: {result.humidity} %\n"
+            "Buzzer: {result.buzzer}\n"
+            "LED brightness: {result.led_brightness}\n"
+            "Child lock: {result.child_lock}\n"
+            "No water: {result.no_water}\n"
+            "Lid opened: {result.lid_opened}\n"
+            "Firmware version: {result.firmware_version}\n"
+            "Hardware version: {result.hardware_version}\n",
+        )
+    )
+    def status(self) -> AirHumidifierStatus:
+        """Retrieve properties."""
+
+        if self.device_info is None:
+            self.device_info = self.info()
```

Comment:
Is this call necessary for functionality?

Reply:
I've used other devices implementation as a reference. This patter is used for humidifiers. `device_info` is lazily initialized, so we need to init it if not done already

Reply:
Ok, fair enough!